test/unit/cxx_test_shmem_atomic_add.cpp

```diff
@@ -36,47 +36,18 @@
 #include <stdio.h>
 #include <shmem.h>
 
-#ifdef ENABLE_SHMEMX_TESTS
-#include <shmemx.h>
-#endif
-
 enum op { ADD = 0, ATOMIC_ADD, CTX_ATOMIC_ADD, FADD, ATOMIC_FETCH_ADD,
-          CTX_ATOMIC_FETCH_ADD, ATOMIC_FETCH_ADD_NBI, CTX_ATOMIC_FETCH_ADD_NBI };
+          CTX_ATOMIC_FETCH_ADD };
 
 #ifdef ENABLE_DEPRECATED_TESTS
-#define DEPRECATED_ADD shmem_add
-#define DEPRECATED_FADD shmem_fadd
+#define DEPRECATED_ADD(TYPENAME, ...) shmem_##TYPENAME##_add(__VA_ARGS__)
+#define DEPRECATED_FADD(TYPENAME, ...) shmem_##TYPENAME##_fadd(__VA_ARGS__)
 #else
-#define DEPRECATED_ADD shmem_atomic_add
-#define DEPRECATED_FADD shmem_atomic_fetch_add
+#define DEPRECATED_ADD(TYPENAME, ...) shmem_##TYPENAME##_atomic_add(__VA_ARGS__)
+#define DEPRECATED_FADD(TYPENAME, ...) shmem_##TYPENAME##_atomic_fetch_add(__VA_ARGS__)
 #endif
 
-#ifdef ENABLE_SHMEMX_TESTS
```

Comment:
@minsii I would like to keep the `SHMEMX_NBI_OPS_CASES` code to make sure we test all the bindings implemented by SOS (the purpose of this test is coverage over the bindings).  This is preserved in the `test/shmemx` code since those files are just being moved.  In general, we like to keep the `test/unit/c11_*`, `test/unit/cxx_*`, and `test/shmemx/cxx_*` code as close as possible to make it easier to maintain.  For example, the generic CXX tests are typically updated by copying the C11 tests and making minor updates so they work with C++.

Reply:
@jdinan Thanks. It looks like the `test/shmemx/cxx*` files in this PR already include NBI AMO cases. I will add the non generic NBI AMO cases in `test/unit/cxx`. I will make my changes based on this PR.